metaserver/src/main/scala/scala/meta/languageserver/compiler/CompletionProvider.scala

```diff
@@ -34,23 +34,23 @@ object CompletionProvider extends LazyLogging {
         items += CompletionItem(
           label = label,
           detail = Some(r.sym.signatureString),
-          kind = symbolKind(compiler)(r)
+          kind = completionItemKind(compiler)(r)
         )
       }
     }
     CompletionList(isIncomplete = false, items = items.result())
   }
 
-  private def symbolKind(c: Global)(r: c.CompletionResult#M): Option[Int] = {
-    if (r.sym.isConstructor) Some(SymbolKind.Constructor)
-    else if (r.sym.isModuleOrModuleClass) Some(SymbolKind.Module)
-    else if (r.sym.isTraitOrInterface) Some(SymbolKind.Interface)
-    else if (r.sym.isClass) Some(SymbolKind.Class)
-    else if (r.sym.isPackageObject) Some(SymbolKind.Namespace)
-    else if (r.sym.isPackage) Some(SymbolKind.Package)
-    else if (r.sym.isMethod) Some(SymbolKind.Method)
-    else if (r.sym.isVal) Some(SymbolKind.Constant)
-    else if (r.sym.isVar) Some(SymbolKind.Variable)
+  private def completionItemKind(c: Global)(r: c.CompletionResult#M): Option[Int] = {
+    if (r.sym.isPackage) Some(CompletionItemKind.Module)
+    else if (r.sym.isModuleOrModuleClass) Some(CompletionItemKind.Module)
+    else if (r.sym.isTraitOrInterface) Some(CompletionItemKind.Interface)
+    else if (r.sym.isClass) Some(CompletionItemKind.Class)
+    else if (r.sym.isPackageObject) Some(CompletionItemKind.Module)
+    else if (r.sym.isMethod) Some(CompletionItemKind.Method)
+    else if (r.sym.isField && r.sym.owner != null && r.sym.owner.isCaseClass) Some(CompletionItemKind.Field)
```

Comment:
Similarly to this special case, how about using `CompletionItemKind.Property` for vals whose owner is not an object?

Reply:
Alternatively, `Field` could be used for type members (projections) and `Property` for this case.

Reply:
I honestly like `Field` for case classes fields. It's quite natural and consistent with my TypeScript experience with VSCode. Can you elaborate on this

> how about using CompletionItemKind.Property for vals whose owner is not an object?

?